middleware/kubernetes/README.md

```diff
@@ -1,88 +1,126 @@
 
-~~~
-kubernetes [ZONES...] {
-    endpoint ENDPOINT
-}
-~~~
+* "1-2-3-4" is the derived from the ip address of the pod
+* "mynamespace" is the k8s namespace for the service, and
+* "pod" indicates this is a pod
+* "coredns.local" is the zone
 
-* **ENDPOINT** the kubernetes API endpoint, defaults to http://localhost:8080
+Endpoint A records are constructed as "epname.myservice.mynamespace.svc.coredns.local" where:
 
-TODO(...): Add all the other options.
+* "epname" is the hostname (or name constructed from IP) of the endpoint
+* "myservice" is the name of the k8s service that the endpoint serves
+* "mynamespace" is the k8s namespace for the service, and
+* "svc" indicates this is a service
+* "coredns.local" is the zone
 
-## Examples
 
-This is the default kubernetes setup, with everything specified in full:
+## Syntax
 
-~~~
-# Serve on port 53
-.:53 {
-    # use kubernetes middleware for domain "coredns.local"
-    kubernetes coredns.local {
-        # Kubernetes data API resync period
-        # Example values: 60s, 5m, 1h
-        resyncperiod 5m
-		
-        # Use url for k8s API endpoint
-        endpoint https://k8sendpoint:8080
-		
-        # The tls cert, key and the CA cert filenames
-        tls cert key cacert
-		
-        # Only expose the k8s namespace "demo"
-        namespaces demo
-		
-        # Only expose the records for kubernetes objects
-        # that match this label selector. The label
-        # selector syntax is described in the kubernetes
-        # API documentation: http://kubernetes.io/docs/user-guide/labels/
-        # Example selector below only exposes objects tagged as
-        # "application=nginx" in the staging or qa environments.
-        #labels environment in (staging, qa),application=nginx
-		
-        # The mode of responding to pod A record requests. 
-        # e.g 1-2-3-4.ns.pod.zone.  This option is provided to allow use of
-        # SSL certs when connecting directly to pods.
-        # Valid values: disabled, verified, insecure
-        #  disabled: default. ignore pod requests, always returning NXDOMAIN
-        #  insecure: Always return an A record with IP from request (without 
-        #            checking k8s).  This option is is vulnerable to abuse if
-        #            used maliciously in conjuction with wildcard SSL certs.
+This is an example kubernetes middle configuration block, with all options described:
+
+```
+# kubernetes <zone> [<zone>] ...
+#
+# Use kubernetes middleware for domain "coredns.local"
+# Reverse domain zones can be defined here (e.g. 0.0.10.in-addr.arpa), 
+# or instead with the "cidrs" option.
+#
+kubernetes coredns.local {
+	
+	# resyncperiod <period>
+	#
+	# Kubernetes data API resync period. Default is 5m
+	# Example values: 60s, 5m, 1h
+	#
+	resyncperiod 5m
+	
+	# endpoint <url>
+	#
+	# Use url for k8s API endpoint. Default is https://localhost:8080
+	#
+	endpoint https://localhost:8080
```

Comment:
If omitted it will try to use the service account (which will only work if it's running in-cluster).